### ISSUE TITLE
Change of statistics method 1

### ISSUE THREAD

**Issue by MichaelaVesterlund** [Sambruk/digitala-produkter-bol#30]
At GR we have found that we are in need of a new field in the call and response for method 1 of statistics. We would like to add orderrowid to be able to match to correct buyer. A schoolUnit can have several buyers and it is important that we can distinguish the licenses.

In the call we would like to add orderrowid, could be in the account object, and that it is optional.
In the response we would like to add it as a field for assignedLicenses and unassignedLicenses.

{
    serviceProviderId:"",
    clientId:"",

    assignedLicenses: [{
        user: {
            identitySource: "",
            id: ""
        },
        **orderRowId: "",**
        licenseKey:"",
	productUrl:"",
        used:true,
        validFrom:"",
        validTo:""
    }],

    unassignedLicenses:[{
        **orderRowId: "",**
        articleNumber:"",
        quantity:10,
        licenseKeys:["", ""]
    }]
}

**Comment by MichaelaVesterlund** [comment]
"However, we have a question regarding the case when "orderRowId" is not provided in the request.
The "unassignedLicenses" part of the response will be split in multiple parts, one part for each unique combination of "orderRowId" and "articleNumber"."

Did you mean when it is provided? When it is provided it should result in a unique combinations of orderrowid and articlenumber. If it is not provided it should be grouped only on articlenumber.

**Comment by MichaelaVesterlund** [comment]
For this to work, our reference/orderrowid has to be saved at the supplier.

**Comment by hekdahl** [comment]
I think it will be difficult to have orderRowId mandatory of two reasons:
1. Reference/orderrowid has to be saved at the supplier, Gleerups dont't save it today
2. We will also return assigned and unassigned licenses not ordered from the current retailer, maybe created in Gleerups, and they might not have an orderRowId

One problem might be that you may have used the same orderRowId as Läromedia. Maybe it is better to add a new field named clientId instead or have both clientId and orderRowId?

You (GR) have to be able to assign not only your own licenses but also licenses from for example Läromedia.

**Comment by MichaelaVesterlund** [comment]
If we would add our ordernumber instead, would that work? In the statistic call, we should only get the licenses we have bought, or do I missunderstand what you mean? In the call we have the field ClientId, which should identify GR. 

We have not talked much about how it should work when we buy from Läromedia. Either they work as a intermediator, or only the order call is made to them, in that case then i think they need to add our ordernumber and clientId, for statistic and assignment calls..?

**Comment by hekdahl** [comment]
@MichaelaVesterlund I guess everyone will save the clientOrderNumber so that probably works better.
As I see it you have to get licenses ordered not only from yourselves. With an BOL-integration with "Tilldelning" it will not be possible for teachers to assign in "Gleerups administration" so every license has to be assigned from your platform.
A Skolon integration solves this by "Gleerups" sending other licenses as distribution orders to them so they can assign them and you have to do the same by retrieving these licenses from us.

**Comment by MichaelaVesterlund** [comment]
What do you mean by other licenses? Do you mean the ones from Läromedia? Should not Läromedia send the licenses to us?

**Comment by hekdahl** [comment]
@MichaelaVesterlund Only one can be the platform (master) that do the assignments of the licenses.
Lets say GR is the master. Then a teacher can put an order from the GR-portal. But another teacher (or the same) can put an order from Läromedia. The order from Läromedia is sent to Gleerups and GR-portal will get information of the order from statistics method 1 so the teacher can assign it in GR-portal.

**Comment by hekdahl** [comment]
@MichaelaVesterlund I correct myself. Yes, in your case the orders might come from Läromedia to you.
Lets say GR is the master. Then a teacher can put an order from the GR-portal. But Gleerups support can add free licenses to a school unit.  GR-portal will get information of these licenses from statistics method 1 so teachers can assign it in GR-portal.

**Comment by MichaelaVesterlund** [comment]
@hekdahl yes, but we would prefer orderLineNumber if that is possible. There is a possibility that an article could be present several times in an order.
